compiler/src/dotty/tools/dotc/transform/init/Summarization.scala

```diff
@@ -59,6 +59,7 @@ object Summarization {
 
         val argTps = fun.tpe.widen match
           case mt: MethodType => mt.paramInfos
+          case _ => Nil // TODO is this the right thing to do?
```

Comment:
@liufengyun is this reasonable or did a macro generate code that should have been rejected by `Ycheck`?

Reply:
I'm wondering if the generated tree is valid? For a typed `Apply` tree, we expect the `fun` has a `MethodType`.

Reply:
All tests passed on the latest nightly build but not on today's master. We might have introduced something today or it could just be the different project setup. We could test this tomorrow with a new nightly build.